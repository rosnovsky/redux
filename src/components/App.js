import React, { useState, useEffect } from 'react';
import '../styles/App.css';
import store from '../store';
import moment from 'moment';
import uuid from 'uuid/v4';

import Headlines from './Headlines';
import News from './News';
import Metadata from './Metadata';
import ReadLater from './ReadLater';

function App() {
    const [articles, setArticles] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        const apiKey = '57ae048d4ae04603bd6bb1973b2cc445';
        const today = moment();
        const date = moment(today)
            .subtract(1, 'weeks')
            .toISOString();
        const excludedDomains =
            'sourceforge.net,github.com,askubuntu.com,stackoverflow.com,pypi.org,digitalocean.com,linuxsecurity.com,linuxgizmos.com,img.tfd.com';
        const url = `https://newsapi.org/v2/everything?q=Linux&from=${date}&language=en&sortBy=publishedAt&excludeDomains=${excludedDomains}&apiKey=${apiKey}`;

        const fetchArticles = async url => {
            try {
                let id = 0;
                setIsLoading(true);
                const response = await fetch(url);
                const data = await response.json();
                const articles = await data.articles;
                articles.forEach(article => {
                    article.id = id;
                    id += 1;
                });
                setArticles(articles);
                store.dispatch({
                    type: 'UPDATE_ARTICLES',
                    articles,
                });
                setIsLoading(false);
                return articles;
            } catch (error) {
                setIsLoading(false);
                throw Error(error);
            }
        };
        fetchArticles(url);
    }, []);

    return (
        <div className="App">
            <h1>Linus News App</h1>
            {<Headlines articles={articles} />}
            <News />
            <Metadata />
            <ReadLater />
        </div>
    );
}

export default App;
