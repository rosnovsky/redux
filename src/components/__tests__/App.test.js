import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import { Provider } from 'react-redux';
import store from '../../store';

describe('App Container', () => {
    it('renders without crashing', () => {
        const app = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        );
        let tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
