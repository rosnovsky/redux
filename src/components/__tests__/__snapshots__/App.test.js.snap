// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container renders without crashing 1`] = `
<div
  className="App"
>
  <h1>
    Linus News App
  </h1>
  <div
    className="App"
  >
    <h2>
      Headlines
    </h2>
  </div>
  <div
    className="App"
  >
    <h2>
      News
    </h2>
  </div>
  <div
    className="App"
  >
    <h2>
      Metadata
    </h2>
  </div>
  <div
    className="App"
  >
    <h4>
      Read Later
    </h4>
  </div>
</div>
`;
